hare/eligibility/oracle.go

```diff
@@ -30,33 +41,63 @@ type VerifierFunc = func(msg, sig, pub []byte) (bool, error)
 
 // Oracle is the hare eligibility oracle
 type Oracle struct {
-	beacon            valueProvider
-	activeSetProvider activeSetProvider
-	vrfSigner         Signer
-	vrfVerifier       VerifierFunc
-	layersPerEpoch    uint16
+	beacon         valueProvider
+	getActiveSet   activeSetFunc
+	vrfSigner      Signer
+	vrfVerifier    VerifierFunc
+	layersPerEpoch uint16
+	cache          casher
 	log.Log
 }
 
 // Returns the relative Layer that w.h.p. we have agreement on its view
 // safe is defined to be k layers prior to the provided Layer
-func safeLayer(layer types.LayerID) types.LayerID {
-	if layer > k { // assuming genesis is zero
-		return layer - k
+func safeLayer(layer types.LayerID, safetyParam types.LayerID) types.LayerID {
+	if layer <= safetyParam { // assuming genesis is zero
+		return config.Genesis
 	}
 
-	return config.Genesis
+	return layer - safetyParam
+}
+
+func roundedSafeLayer(layer types.LayerID, safetyParam types.LayerID,
```

Comment:
is this part well tested?

Reply:
Well, I added UT for it. Unless you see something in specific I think it is well tested.